docs/data/data-grid/ai-assistant/ai-assistant.md

```diff
@@ -0,0 +1,162 @@
+
+`mockPromptResolver()` can handle a predefined set of prompts:
+
+- `sort by name`
+- `sort by company name and employee name`
+- `show people from the EU`
+- `order companies by amount of people`
+
+You can use suggestions to quickly enter prompts that are supported by the mock resolver.
+
+:::
+
+## Custom examples
+
+Use the `examples` prop in the `columns` array to provide custom examples as context for prompt processing.
+The `examples` prop should contain an array of possible values for its respective column.
+
+{{"demo": "AssistantWithExamples.js", "bg": "inline"}}
+
+## Use row data for examples
+
+Pass the `allowAiAssistantDataSampling` prop to use row data to generate examples.
+This is useful if you're dealing with non-sensitive data and want to skip creating custom examples for each column.
+
+Data is collected randomly at the cell level, which means that the examples for a given column might not come from the same rows.
+
+For the `aiAssistant.getPromptContext()` API method, pass the `allowDataSampling` flag as a parameter.
+
+```ts
+const context = React.useMemo(
+  () => apiRef.current.aiAssistant.getPromptContext(allowDataSampling),
+  [apiRef, allowDataSampling],
+);
+```
+
+{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}
+
+## Using Server-side data
+
+The example below shows how to combine prompt control with [server-side data](/x/react-data-grid/server-side-data/).
+
+{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}
+
+## Processing service integration
+
+Natural language prompts must be processed by a service to understand what kinds of state changes must be applied to the Data Grid to match the user's request.
+You can use MUI's processing service or build your own.
+
+### With MUI's service
+
+The Data Grid provides all the necessary elements for integration with MUI's service.
+
+1. Enable the AI Assistant feature by adding the `enableAiAssistant` prop.
+   A new toolbar button will appear in the default [`<Toolbar />`](/x/react-data-grid/components/toolbar/).
+   This button opens `<AssistantPanel />` that can take user's prompts.
+2. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.
+
+   :::error
+   Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request on to MUI's service.
+
+   This is an example of a [Fastify proxy](https://www.npmjs.com/package/@fastify/http-proxy) for the prompt requests.
+
+   ```ts
+   fastify.register(proxy, {
+     upstream: 'https://api.mui.com',
+     prefix: '/api/my-custom-path',
+     rewritePrefix: '/api/v1/datagrid/prompt',
+     replyOptions: {
+       rewriteRequestHeaders: (_, headers) => ({
+         ...headers,
+         'x-api-key': process.env.MUI_DATAGRID_API_KEY,
+       }),
+     },
+   });
+   ```
+
+   :::
+
+3. Provide the `onPrompt()` callback to pass the user's prompts to the service.
```

Comment:
I assume that an end-to-end example isn’t feasible in this case. Given that, I think it would be helpful for users to have a basic starter template repo that includes the proxy setup, allowing them to easily clone and integrate it with their backend of choice.

Would it make sense to consider this as a potential follow-up action for this PR?

Reply:
Yeah, we can evaluate the effort on a followup. 
Now that we have the model working for the examples, users can play with it already.